drivers/mtd_sdmmc/mtd_sdmmc.c

```diff
@@ -0,0 +1,254 @@
+/*
+ * Copyright (C) 2017 HAW-Hamburg
+ *               2023 Gunar Schorcht
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ *
+ */
+
+/**
+ * @ingroup     drivers_mtd_sdmmc
+ * @{
+ *
+ * @file
+ * @brief       Driver for using sdmmc via mtd interface
+ *
+ * @author      Michel Rottleuthner <michel.rottleuthner@haw-hamburg.de>
+ * @author      Gunar Schorcht <gunar@schorcht.net>
+ *
+ * @}
+ */
+
+#include <errno.h>
+#include <inttypes.h>
+#include <string.h>
+
+#define ENABLE_DEBUG 0
+#include "debug.h"
+#include "kernel_defines.h"
+#include "macros/utils.h"
+#include "mtd.h"
+#include "mtd_sdmmc.h"
+#include "sdmmc/sdmmc.h"
+
+static int mtd_sdmmc_init(mtd_dev_t *dev)
+{
+    DEBUG("mtd_sdmmc_init\n");
+    mtd_sdmmc_t *mtd_sd = (mtd_sdmmc_t*)dev;
+
+    /* get the SDMMC device descriptor from SDMMC peripheral index */
+    mtd_sd->sdmmc = sdmmc_get_dev(mtd_sd->sdmmc_idx);
```

Comment:
Do we really need that `sdmmc_get_dev()` function? We could just as well set `mtd_sd->sdmmc` on declaration instead of `sdmmc_idx`

Reply:
> Do we really need that `sdmmc_get_dev()` function?

Do you mean in general or in `mtd_sdmmc.c`?

Reply:
The low-level test `tests/drivers/sdmmc` needs it definitely to get the `sdmmc_dev` for the given device index. For example in command `init 1`.

Reply:
My main concern was that this would make it harder when there are multiple sdmmc backends (e.g. `periph_sdmmc` and `sdcard_spi`) - and designing an API just to make the test easier isn't something that should be done when it makes the non-test case more complex.

I don't think it makes sense to use sdmmc without MTD.

Reply:
> My main concern was that this would make it harder when there are multiple sdmmc backends (e.g. `periph_sdmmc` and `sdcard_spi`)

Ah ok, I see :thinking: 

> I don't think it makes sense to use sdmmc without MTD.

What about SDIO devices that will be supported in future? An SDMMC device must not necessarily a MTD.

Reply:
> What about SDIO devices that will be supported in future? An SDMMC device must not necessarily a MTD.

Good point. That would also mean that `sdcard_spi` can't be an interchangeable backend with `periph_sdmmc` anyway as it only works for SPI mode.

In that case we would only ever have two SD card implementations (not accounting for legacy drivers that could be ported to `periph_sdmmc`) and there wouldn't ever be multiple implementations of `periph_sdmmc` on the same platform.

Reply:
Yea let's deal with that if this need ever arises, from what it looks like this will only ever deal with in-MCU implementations in the forseeable future.